### PR TITLE
Add tenantID to suggested app token cache key

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs [hunk 1]
```diff
@@ -54,8 +55,8 @@ private static bool GetOboOrAppKey(AuthenticationRequestParameters requestParame
 
             if (requestParameters.ApiId == TelemetryCore.Internal.Events.ApiEvent.ApiIds.AcquireTokenForClient)
             {
-                key = requestParameters.AppConfig.ClientId + "_AppTokenCache";
-                return true;
+                string tenantId = requestParameters.Authority.TenantId ?? "";
+                key = $"{requestParameters.AppConfig.ClientId}_{tenantId}_AppTokenCache";
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs:59]: Test scenarios: 
- authority is "l.m.o/common" -> tenant id is common
- authority is tenanted -> tenant id
- authority does not support tenants (ADFS or B2C) -> ""

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs:59]: Thanks @yohanb for proposing the fix!

File: src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs [hunk 2]
```diff
@@ -54,8 +55,8 @@ private static bool GetOboOrAppKey(AuthenticationRequestParameters requestParame
 
             if (requestParameters.ApiId == TelemetryCore.Internal.Events.ApiEvent.ApiIds.AcquireTokenForClient)
             {
-                key = requestParameters.AppConfig.ClientId + "_AppTokenCache";
-                return true;
+                string tenantId = requestParameters.Authority.TenantId ?? "";
```
Comment (yohanb) [inline src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs:58]: you're not using the tenantId var in your returned result

Comment (yohanb) [reply src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs:58]: also if tenantId is an empty string, key will have double underscores, i.e. clientId__AppTokenCache

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs:58]: This is fine, I think it's more readable this way, to show that a tenantId does not exist (i.e. ADFS authority)

**Review by yohanb — APPROVED** [review]

**Comment by yohanb** [comment]
awesome, thanks @bgavrilMS !  What is the release cycle like and when can I expect to be able to use this?

**Comment by bgavrilMS** [comment]
We expect to release MSAL 4.26 by mid next-week. Sooner if testing goes well.

**Pull request merged**